File: BOJ/Brute_Force/BJ2231.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim();

const N = parseInt(input);
let flag = false;

for(let i=1; i<=N; i++) {
  let sum=0;
  let temp = i;

  while(temp>0) {
    sum += temp % 10;
    temp = Math.floor(temp/10);
  }
  if(N === sum + i) {
    flag = true;
    console.log(i);
    break;
  }
}
if(!flag) console.log(0);

File: BOJ/Brute_Force/BJ2231.cpp
#include <iostream>
using namespace std;

int N;
bool flag;

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  cin >> N;

  for(int i=1; i<=N; i++) {
    int sum = 0;
    int temp = i;

    while(temp > 0) {
      sum += temp % 10;
      temp /= 10;
    }
    
    if(i+sum == N) {
      cout << i << '\n';
      flag = true;
      break;
    }
  }
  if(!flag) cout << '0' << '\n';
  return 0;
}